src/libs/actions/Task.js

```diff
@@ -0,0 +1,102 @@
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+
+/**
+ * Assign a task to a user
+ * @param {String} parentReportActionID
+ * @param {String} parentReportID
+ * @param {String} taskReportID
+ * @param {String} name
+ * @param {String} description
+ * @param {String} assignee
+ * @param {String} ownerEmail
+ *
+ */
+
+function assignTask(parentReportActionID, parentReportID, taskReportID, name, description, assignee, assigneeChatReportID, ownerEmail) {
+    let parentReport;
+
+    // If there isn't a parentReportId provided and assignee is given, we need to optimistically create a new parent chat report
+    // between the task creator and the assignee.
+    if (!parentReportID && assignee) {
+        parentReport = ReportUtils.getChatByParticipants([assignee]);
+    }
+
+    if (!parentReportID && !parentReport && assignee) {
+        parentReport = ReportUtils.buildOptimisticChatReport([assignee]);
+    }
```

Comment:
Since this falls into the flow where we need to create an optimistic parent chat report, let's add it in the above if statement